### DISCUSSION TITLE
updating hibernate-orm to 6.1

### DISCUSSION THREAD

**Discussion by nicola-baechi** [quarkusio/quarkus#28900] [Q&A] [answered]
I'm using version 2.13.3 of quarkus and noticed that the hibernate-orm version is 5.6.11. I require the feature to use subqueries in the FROM clause which was introduced in 6.1 (see [this article](https://in.relation.to/2022/06/24/hibernate-orm-61-features/)). Is there any chance it will be updated soon? And if not, is there any way to update the dependency manually without having to rely on a new release of quarkus?

[a comment by an automated account is not shown]

**Comment by gsmet** [comment, marked as answer]
The upgrade to Hibernate ORM 6 is part of the Jakarta effort we have started as it is targeting JPA 3.1. The work has started but it not completed yet as ORM 6 changes a lot of stuff that requires adjustments in Quarkus.

You can't upgrade to ORM 6 with Quarkus 2.x. But keep in mind that 5.6 is actively maintained.

**Reply by mabartos** [reply]
@gsmet Hi! Do you have any ETA of Hibernate ORM 6 available in Quarkus 3 pre-releases?

**Reply by Sanne** [reply]
@mabartos that's a sore point :)
We need both Hibernate ORM 6 and Hibernate Reactive upgraded to match Hibernate ORM 6, otherwise reactive applications would no longer work. The Hibernate Reactive team is working on this but there's quite some work to be done yet - I optimistically hope we'll be done in a ~1 month but it's hard to say for sure at this stage.

**Reply by mabartos** [reply]
@Sanne Ok, thanks for the info.

**Reply by mabartos** [reply]
@gsmet @Sanne Hello :)) Any additional info about Hibernate ORM 6 integration with Quarkus 3.x?

**Reply by Sanne** [reply]
Hi @mabartos - Hibernate Reactive 2.0 is getting closer but it's not fully functional yet; roughly 80% of integration tests are passing now - there's good velocity but we had some setbacks, it's fairly tricky work and we need to get it right.
